### PR TITLE
[agent] add mainloop poll timeout config

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM

File: src/agent/application.cpp [hunk 1]
```diff
@@ -44,8 +44,12 @@
 
 namespace otbr {
 
+#ifndef OTBR_MAINLOOP_POLL_TIMEOUT
+#define OTBR_MAINLOOP_POLL_TIMEOUT 10
```
Comment (superwhd) [inline src/agent/application.cpp:48]: nit: mention the time unit in the name

File: src/agent/application.cpp [hunk 2]
```diff
@@ -44,8 +44,12 @@
...
+#endif
```
Comment (bukepo) [inline src/agent/application.cpp:49]:
```suggestion
#ifndef OTBR_MAINLOOP_POLL_TIMEOUT_SEC
#define OTBR_MAINLOOP_POLL_TIMEOUT_SEC 10
#endif
```

File: src/agent/application.cpp [hunk 3]
```diff
@@ -44,8 +44,12 @@
...
+
 std::atomic_bool     Application::sShouldTerminate(false);
-const struct timeval Application::kPollTimeout = {10, 0};
+const struct timeval Application::kPollTimeout = {OTBR_MAINLOOP_POLL_TIMEOUT, 0};
```
Comment (bukepo) [inline src/agent/application.cpp:52]:
```suggestion
const struct timeval Application::kPollTimeout = {OTBR_MAINLOOP_POLL_TIMEOUT_SEC, 0};
```

**Review by superwhd — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**